src/reformat.c

```diff
@@ -109,50 +109,60 @@ avifBool avifPrepareReformatState(const avifImage * image, const avifRGBImage *
     state->rgbMaxChannel = (1 << rgb->depth) - 1;
     state->yuvMaxChannelF = (float)state->yuvMaxChannel;
     state->rgbMaxChannelF = (float)state->rgbMaxChannel;
-    state->uvBias = 1 << (image->depth - 1);
+    state->yBias = state->yuvRange == AVIF_RANGE_LIMITED ? (float)(16 << (state->yuvDepth - 8)) : 0.f;
+    state->uvBias = (float)(1 << (state->yuvDepth - 1));
+    state->yRange = state->yuvRange == AVIF_RANGE_LIMITED ? (float)(219 << (state->yuvDepth - 8)) : state->yuvMaxChannelF;
+    state->uvRange = state->yuvRange == AVIF_RANGE_LIMITED ? (float)(224 << (state->yuvDepth - 8)) : state->yuvMaxChannelF;
 
     uint32_t cpCount = 1 << image->depth;
     for (uint32_t cp = 0; cp < cpCount; ++cp) {
-        int unormY = cp;
-        int unormUV = cp;
-        if (image->yuvRange == AVIF_RANGE_LIMITED) {
-            unormY = avifLimitedToFullY(image->depth, unormY);
-            unormUV = avifLimitedToFullUV(image->depth, unormUV);
-        }
-        state->unormFloatTableY[cp] = (float)unormY / state->yuvMaxChannelF;
-        if (state->mode == AVIF_REFORMAT_MODE_IDENTITY) {
-            state->unormFloatTableUV[cp] = (float)unormY / state->yuvMaxChannelF;
-        } else {
-            state->unormFloatTableUV[cp] = ((float)unormUV - state->uvBias) / state->yuvMaxChannelF;
+        int unorm = cp;
+        switch (state->mode) {
+            // review this when implementing YCgCo limited range support.
+            case AVIF_REFORMAT_MODE_YCGCO:
+            case AVIF_REFORMAT_MODE_YUV_COEFFICIENTS:
+                state->unormFloatTableY[cp] = ((float)unorm - state->yBias) / state->yRange;
+                state->unormFloatTableUV[cp] = ((float)unorm - state->uvBias) / state->uvRange;
+                break;
+            case AVIF_REFORMAT_MODE_IDENTITY:
+                state->unormFloatTableY[cp] = ((float)unorm - state->yBias) / state->yRange;
+                state->unormFloatTableUV[cp] = ((float)unorm - state->yBias) / state->yRange;
+                break;
         }
     }
     return AVIF_TRUE;
 }
 
+// Formulas 20-31 from https://www.itu.int/rec/T-REC-H.273-201612-I/en
```

Comment:
Nit: Formulas => Equations

Reply:
Word "Formulas" was used at line 531 and 536 and I'm following that. Do we want to change these usages as well?

Reply:
Ah, I didn't know that. No need to change this then. (I suggested "Equations" only because that's the term in H.273.)